niworkflows/interfaces/utils.py

```diff
@@ -77,3 +110,25 @@ def _run_interface(self, runtime):
         self._results['out_file'] = os.path.abspath("motion_params.txt")
         np.savetxt(self._results['out_file'], mpars)
         return runtime
+
+
+def _copyxform(ref_image, out_image, message=None):
+    # Read in reference and output
+    resampled = nb.load(out_image)
+    orig = nb.load(ref_image)
+
+    if orig.affine != resampled.affine:
```

Comment:
The problem on poldracklab/fmriprep#710 was exactly that the affines matched, but at certain steps, both ANTs and FSL would change the x-forms. But the affines still matched.

This is also to answer the previous comment: we need to make sure that xforms are not changed by accident. The rationale behind this and the accompanying PR in fmriprep is that:

1. When moving within the same space (SDC, head motion correction, SBRef-EPI), we keep the original xforms (even if they are wrong in the original files).

2. When moving to different space (anatomical or MNI), we are generating an image taken as reference space. So it is this point when we need to make sure that the xform code is the appropriate one. Since we pass on this reference image to the registration interface, the xform matrices should, indeed, be copied over the resulting images. Same for ApplyTransforms.

Bottomline: no interface should change xforms unless we are certain we want to do that. Only [`GenerateSamplingReference`](https://github.com/poldracklab/fmriprep/blob/master/fmriprep/interfaces/images.py#L38) should make these changes.

Reply:
If the affines match, what are we correcting? Is it that the sforms matched but the qforms didn't?